### PR TITLE
Add rounded style on navigation component.

### PR DESCRIPTION
**Pull request by CuteLeon**
# Description

Add rounded-r-xl style on navigation component.

## Issue reference
null

## Checklist

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation

### CODE REVIEW

**Comment by doddgu** [comment]
Thanks for your PR. Please provide the rounded navigation screenshot. This will improve the review efficiency.
And why you set it rounded? We designed a outer border for it.

**Comment by CuteLeon** [comment]
image
Hi @doddgu , screenshot attached.
This improvement is used to make the style of navigation component more uniform with other components.

**Comment by doddgu** [comment]
@CuteLeon Hi, flat is by design. If you like rounded, you can add a settings for it.

**Comment by CuteLeon** [comment]
Hi, @doddgu, I Implemented a NavigationStyle config to control navigation bar style. And kept default option as current Flat approuch.
image

**Review by doddgu — APPROVED** [review]
LGTM. @capdiem PTAL

**Comment by capdiem** [comment]
@CuteLeon Indent 8 spaces?

**Comment by CuteLeon** [comment]
My IDE replaced white-space characters automatically, I have reverted them;

> @CuteLeon Indent 8 spaces?

File: Masa.Blazor.Pro/Shared/Settings.razor [hunk 1]
```diff
@@ -34,13 +41,21 @@
...
+    [Parameter]
+    public EventCallback<string> NavigationBarStyleChanged { get; set; }
+
     protected override async Task OnInitializedAsync()
     {
         if (PageModelChanged.HasDelegate)
             await PageModelChanged.InvokeAsync(GlobalConfig.PageMode);
+        if (NavigationBarStyleChanged.HasDelegate)
+            await NavigationBarStyleChanged.InvokeAsync(GlobalConfig.NavigationStyle);
     }
```
Comment (capdiem) [inline Masa.Blazor.Pro/Shared/Settings.razor:59]: Is this necessary?

File: Masa.Blazor.Pro/Shared/Navigation.razor [hunk 2]
```diff
@@ -2,7 +2,7 @@
 @inject NavHelper NavHelper
 @inject GlobalConfig GlobalConfig;
 
-<MNavigationDrawer @bind-Value="Visible" Class="navigation" Width=300 MiniVariantWidth=80 Fixed ExpandOnHover="GlobalConfig.ExpandOnHover" @bind-MiniVariant="GlobalConfig.NavigationMini" App>
+<MNavigationDrawer @bind-Value="Visible" Class="@($"navigation {(GlobalConfig.NavigationStyle==NavigationStyles.Rounded?"rounded-r-xl":string.Empty)}")" Width=300 MiniVariantWidth=80 Fixed ExpandOnHover="GlobalConfig.ExpandOnHover" @bind-MiniVariant="GlobalConfig.NavigationMini" App>
```
Comment (capdiem) [inline Masa.Blazor.Pro/Shared/Navigation.razor:5]:
```suggestion
<MNavigationDrawer @bind-Value="Visible" Class="@($"navigation {ComputedNavigationStyle}")" Width=300 MiniVariantWidth=80 Fixed ExpandOnHover="GlobalConfig.ExpandOnHover" @bind-MiniVariant="GlobalConfig.NavigationMini" App>
```

**Review by capdiem — CHANGES_REQUESTED** [review]

File: Masa.Blazor.Pro/Shared/Settings.razor (hunk 1 above)
Comment (CuteLeon) [reply Masa.Blazor.Pro/Shared/Settings.razor:59]: Yup, read current navigation style from cache when initializing.

**Review by capdiem — APPROVED** [review]
LGTM

**Comment by pccai** [comment]
Online Demo cannot setting rounded style？

**Comment by capdiem** [comment]
@pccai let me check

**Comment by capdiem** [comment]
@pccai Updated.

**Pull request merged**